lib/salus/scanners/semgrep.rb

```diff
@@ -127,51 +95,82 @@ def run
                 msg: match['message']
               }
             end
-            begin
-              # parse the output
-              output_data = JSON.parse(shell_return.stdout)
-              error_str = messages_str_from_errors(output_data["errors"])
-              # only take the first line of stderror because the other lines
-              # are verbose debugging info generated based on a temp file
-              # so the filename is random and fails the test.
-
-              errors << {
-                status: shell_return.status,
-                stderr: (shell_return.stderr.split("\n").first || "") \
-                + "\n\n" + error_str
-              }
-            rescue JSON::ParserError
-              # only take the first line of stderror because the other lines
-              # are verbose debugging info generated based on a temp file
-              # so the filename is random and fails the test.
-              errors << {
-                status: shell_return.status,
-                stderr: shell_return.stderr.split("\n").first
+          else
+            hits.each do |hit|
+              msg = message_from_hit(hit, match)
+              if match["forbidden"]
+                failure_messages << "\nForbidden #{user_message} was found " \
+                "- #{msg}\n" \
+                "\t#{hit_to_string(hit, base_path)}"
+              end
+              all_hits << {
+                pattern: match['pattern'],
+                config: match['config'],
+                forbidden: match["forbidden"],
+                required: match["required"],
+                msg: msg,
+                hit: hit_to_string(hit, base_path)
               }
             end
-          else
+          end
+
+        # possible exit codes from https://github.com/returntocorp/semgrep/blob/9ac58092cb8ac02bb1f41f59808d4f03a5b8206e/semgrep/semgrep/util.py#L11-L18
+        elsif [1, 2, 3, 4, 5, 6, 7].include?(shell_return.status)
```

Comment:
Following can be used as short-cut:

(1..7).to_a.include?(3)

Reply:
Thanks Saikat, I'll make the update, agree it's clearer. 👍 

This was existing code that git diff is picking up from unwinding the Dir.chdir(@repository.path_to_repo) do block.